purchases/src/main/kotlin/com/revenuecat/purchases/VirtualCurrencyInfo.kt

```diff
@@ -0,0 +1,40 @@
+package com.revenuecat.purchases
+
+import android.os.Parcelable
+import com.revenuecat.purchases.models.RawDataContainer
+import com.revenuecat.purchases.utils.JSONObjectParceler
+import kotlinx.parcelize.IgnoredOnParcel
+import kotlinx.parcelize.Parcelize
+import kotlinx.parcelize.TypeParceler
+import org.json.JSONObject
+
+@Parcelize
+@TypeParceler<JSONObject, JSONObjectParceler>()
+data class VirtualCurrencyInfo(
+    val balance: Long,
+    private val jsonObject: JSONObject,
+) : Parcelable, RawDataContainer<JSONObject> {
+
+    @IgnoredOnParcel
+    override val rawData: JSONObject
+        get() = jsonObject
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+
+        other as VirtualCurrencyInfo
+        return balance == other.balance
+    }
+
+    override fun hashCode(): Int {
+        return balance.hashCode()
+    }
+
+    companion object {
+        fun fromJson(json: JSONObject): VirtualCurrencyInfo {
+            val balance = json.getLong("amount")
```

Comment:
The value returned by the server may change in the future to "balance" in the future - we'll update that in a follow-up PR if/when it happens.